src/codecs/gif.rs

```diff
@@ -154,7 +168,17 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for GifDecoder<R> {
         } else {
             // If the frame does not match the logical screen, read into an extra buffer
             // and 'insert' the frame from left/top to logical screen width/height.
-            let mut frame_buffer = vec![0; self.reader.buffer_size()];
+            let buffer_size = self.reader.buffer_size();
+
+            self.limits.reserve_usize(buffer_size)?;
+
+            let mut frame_buffer = vec![0; buffer_size];
+
+            self.limits.free(
+                u64::try_from(buffer_size)
+                    .expect("if buffer_size overflows a usize, we should have returned already"),
```

Comment:
I don't think this is accurate. [This path](https://github.com/image-rs/image/pull/1696/files#diff-9f2f0567fa73b9d4104f217bfe7c08f7a3eaba4a55e7eba3652e39b041a29d83R136-R138) did not return but `buffer_size` was invalid as `u64`. Should `reserve_usize` return `ImageResult<Option<u64>>`?

Reply:
ah right

hmm..

Could make a `free_usize` which does the u64::try_from().unwrap() but only if a limit is set? And otherwise it's a no-op? Since we should have returned already if a limit is set, but yeah, we might not have if no limit was set.

Reply:
Okay, I tried free_usize since it keeps the calling code simpler. `reserve_usize` returning an `Option<u64>` would mean you need to store it and then conditionally `free`. Happy with the current API for now.